bubblewrap.c

```diff
@@ -1443,6 +1452,36 @@ setup_newroot (bool unshare_pid,
 
           break;
 
+        case SETUP_MOUNT_SYS:
+          if (ensure_dir (dest, 0755) != 0)
+            die_with_error ("Can't mkdir %s", op->dest);
+
+          if (unshare_pid || opt_pidns_fd != -1)
+            {
+              /* Our own sysfs */
+              privileged_op (privileged_op_socket,
+                             PRIV_SEP_OP_SYS_MOUNT, 0, 0, 0,
+                             dest, NULL);
+
+              /* In case the host utilizes SELinux, /sys/fs/selinux should be shared with the sandbox */
+              char *selinux_src_dir = "oldroot/sys/fs/selinux";
+              cleanup_free char *selinux_dest_dir = strconcat (dest, "/fs/selinux");
+              if (stat (selinux_src_dir, &sbuf) == 0 && stat (selinux_dest_dir, &sbuf) == 0)
+                {
+                  privileged_op (privileged_op_socket,
+                                 PRIV_SEP_OP_BIND_MOUNT, 0, 0, 0,
+                                 selinux_src_dir, selinux_dest_dir);
+                }
+            }
+          else
+            {
+              /* Use system sysfs, as we share pid namespace anyway */
+              privileged_op (privileged_op_socket,
+                             PRIV_SEP_OP_BIND_MOUNT, 0, 0, 0,
+                             "oldroot/sys", dest);
+            }
```

Comment:
This seems quite "do what I mean". I think it would be easier to document this option clearly if it *always* created a new sysfs instance, and never bind-mounted the existing `/sys`.

(Also easier to be sure that it's correct - if the option autodetects whether it needs a new instance or can keep the current instance, then that opens the possibility that the autodetection has bugs.)